### PR TITLE
ID-11

### PR DESCRIPTION
**Pull request by xiangzhe0812**

### CODE REVIEW

File: app/screens/claim/index.tsx [hunk 1]
```diff
@@ -103,34 +138,130 @@ const Claim = () => {
...
+  textStyle: {
+    color: "white",
+    fontWeight: "bold",
+    textAlign: "center",
+  },
+  modalText: {
+    marginBottom: 15,
+    textAlign: "center",
+  },
+});
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:267]: Can we move this to its own file?

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
One suggestion

File: app/screens/claim/index.tsx [hunk 2]
```diff
@@ -103,34 +138,130 @@ const Claim = () => {
...
+        );
       case "Email":
         return <EmailClaim item={claim} />;
       case "Mobile":
         return <MobileClaim item={claim} />;
       case "18+":
         return (
           <>
-          <SelectClaim
+            <SelectClaim
```
Comment (ahinchliff) [inline app/screens/claim/index.tsx:154]: delete <>

File: app/screens/claim/index.tsx [hunk 3]
```diff
@@ -103,34 +138,130 @@ const Claim = () => {
...
-      {renderClaim(claim.type || "")}
-    </View>
+    <>
+      <View style={styles.claim.root}>
+        <Text style={styles.claim.title}>{claim?.type}</Text>
+        <Text style={styles.claim.label}>{claim?.description}</Text>
+        {renderClaim(claim.type || "")}
+      </View>
+
+      {modalVisible ? (
```
Comment (ahinchliff) [inline app/screens/claim/index.tsx:182]: { modalVisible && (

File: app/screens/claim/index.tsx [hunk 4]
```diff
@@ -103,34 +138,130 @@ const Claim = () => {
...
+            <View style={styless.centeredView}>
+              <View style={styless.modalView}>
+                <Text style={styless.modalText}>
+                  Are you sure, you want to select this image/document for
+                  claims?{" "}
+                </Text>
+                <View style={{ display: "flex", flexDirection: "row" }}>
+                  <Pressable
+                    style={[styless.button, styless.buttonClose]}
+                    onPress={() => claimsDocuments()}
```
Comment (ahinchliff) [inline app/screens/claim/index.tsx:202]: onPress={claimsDocuments}

File: app/screens/claim/index.tsx [hunk 5]
```diff
@@ -103,34 +138,130 @@ const Claim = () => {
...
+                  >
+                    <Text style={styless.textStyle}>Cancel</Text>
+                  </Pressable>
+                </View>
+              </View>
+            </View>
+          </Modal>
+        </View>
+      ) : (
+        false
```
Comment (ahinchliff) [inline app/screens/claim/index.tsx:218]: dont need this

File: app/screens/claims/index.tsx [hunk 6]
```diff
@@ -78,11 +77,15 @@ const ClaimSelector = () => {
   return (
     <View style={styles.list.root as ViewStyle}>
       <StatusBar barStyle="light-content" />
-      {assetStore.claims[0] && <Profile fullName={assetStore.claims.find((c => c.key === "0x02"))!} emailAddress={assetStore.claims.find((c => c.key === "0x03"))!} />}
+      {assetStore.claims && assetStore.claims.length > 0 && (
```
Comment (ahinchliff) [inline app/screens/claims/index.tsx:80]: assetStore?.claims.length

**Review by ahinchliff — CHANGES_REQUESTED** [review]
Just a few small nit picky things

File: app/screens/claim/index.tsx [hunk 7]
```diff
@@ -28,71 +45,89 @@ const Claim = () => {
...
-      const libraryUrl = await AsyncStorage.getItem("library_url");
-      if(libraryUrl === null) {
-        let result = await DocumentPicker.getDocumentAsync();
-        if (result.type === "success") {
-          await AsyncStorage.setItem("document_url", JSON.stringify(await getData(result)));
-        }
-      } else {
-        // TODO waiting for message
-        alert('You have already selected one document');
+      let result = await DocumentPicker.getDocumentAsync();
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:55]: @ahinchliff be good if you could also make the suggestion like this

```suggestion
      const result = await DocumentPicker.getDocumentAsync();
```

File: app/screens/claim/index.tsx [hunk 8]
```diff
@@ -28,71 +45,89 @@ const Claim = () => {
...
-          aspect: [4, 3],
-          quality: 1,
-        })
-        if (!result.cancelled) {
-          await AsyncStorage.setItem("library_url", JSON.stringify(await getData(result)));
-        }
-      }else {
-        // TODO waiting for message
-        alert('You have already selected one document');
+      let result = await ImagePicker.launchImageLibraryAsync({
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:72]:
```suggestion
      const result = await ImagePicker.launchImageLibraryAsync({
```

**Review by rellfy — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

**Pull request merged**